core/src/main/java/com/linecorp/armeria/server/VirtualHost.java

```diff
@@ -195,10 +201,41 @@ private static boolean needsNormalization(String hostnamePattern) {
     }
 
     @Nullable
-    static SslContext validateSslContext(@Nullable SslContext sslContext) {
+    static SslContext validateSslContext(@Nullable SslContext sslContext) throws SSLException {
         if (sslContext != null && !sslContext.isServer()) {
             throw new IllegalArgumentException("sslContext: " + sslContext + " (expected: server context)");
         }
+
+        SSLEngine serverEngine = null;
+        SSLEngine clientEngine = null;
+
+        try {
+            serverEngine = sslContext.newEngine(ByteBufAllocator.DEFAULT);
+            serverEngine.setUseClientMode(false);
+            serverEngine.setNeedClientAuth(false);
+
+            final SslContext sslContextClient =
+                    VirtualHostBuilder.buildSslContext(SslContextBuilder::forClient, sslContextBuilder -> {});
+            clientEngine = sslContextClient.newEngine(ByteBufAllocator.DEFAULT);
+            clientEngine.setUseClientMode(true);
+
+            clientEngine.beginHandshake();
+            serverEngine.beginHandshake();
+
+            final ByteBuffer appBuf = ByteBuffer.allocate(clientEngine.getSession().getApplicationBufferSize());
+            final ByteBuffer packetBuf = ByteBuffer.allocate(clientEngine.getSession().getPacketBufferSize());
+
+            clientEngine.wrap(appBuf, packetBuf);
+            appBuf.clear();
+            packetBuf.flip();
+            serverEngine.unwrap(packetBuf, appBuf);
+        } catch (SSLException e) {
+            throw new SSLException("failed to validate SSL/TLS configuration: " + e, e);
```

Comment:
I think we don't have to put `e` two times. How about replacing with `sslContext`?
```
throw new SSLException("failed to validate SSL/TLS configuration: " + sslContext, e);
```

Reply:
I'm not sure. `SslContext.toString()` does not always yield a good hint. The intention of appending `e` here is to make it easy to find the cause of the validation failure easily. However, I don't think we need to repeat the exception type. How about:

    "failed to ...." + e.getMessage()

Reply:
I agree,
* Users need useful message to find the cause easily.
* We don't need to repeat the exception type.

I modified to 
``` java
throw new SSLException("failed to validate SSL/TLS configuration: " + e.getMessage());
```